stress/src/throughput.rs

```diff
@@ -0,0 +1,51 @@
+use rayon::prelude::*;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+
+const SLIDING_WINDOW_SIZE: u64 = 2; // In seconds
+
+static STOP: AtomicBool = AtomicBool::new(false);
+
+pub fn test_throughput<F>(func: F)
+where
+    F: Fn() + Sync + Send + 'static,
+{
+    ctrlc::set_handler(move || {
+        STOP.store(true, Ordering::SeqCst);
+    })
+    .expect("Error setting Ctrl-C handler");
+    let mut start_time = Instant::now();
+    let mut end_time = Instant::now();
+    let mut total_count_old: u64 = 0;
+    let total_count = Arc::new(AtomicU64::new(0));
+    let total_count_clone = Arc::clone(&total_count);
+
+    rayon::spawn( move || {
+        (0..num_cpus::get()).into_par_iter().for_each(|_| loop {
```

Comment:
for a follow up PR - make the count configurable instead of defaulting to num_cpus. We can use it to measure how good we scale by trying different thread count from 1, 2, ...till num_cpus.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/1055